src/jit/regset.cpp

```diff
@@ -1535,8 +1538,15 @@ void RegSet::rsSpillTree(regNumber reg, GenTreePtr tree, unsigned regIdx /* =0 *
         ReturnTypeDesc* retTypeDesc = call->GetReturnTypeDesc();
         treeType                    = retTypeDesc->GetReturnRegType(regIdx);
     }
+#ifdef _TARGET_ARM_
+    else if (tree->OperIsPutArgSplit())
+    {
+        splitArg = tree->AsPutArgSplit();
+        treeType = splitArg->GetRegType(regIdx);
+    }
     else
```

Comment:
This line is now under `#ifdef _TARGET_ARM_`, so for the non-arm case what is supposed to be the else clause of the `if (tree->IsMultiRegCall())` is unconditionally executed, which could possibly explain the x86 test failure.